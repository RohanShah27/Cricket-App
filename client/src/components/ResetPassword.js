import React, { Component } from "react";
import { resetPassword } from "../actions/User";
import { connect } from "react-redux";
import "../styles/resetpassword.css";
export class ResetPassword extends Component {
  constructor(props) {
    super(props);
    this.onSubmitClick = this.onSubmitClick.bind(this);
  }
  state = {
    //values in the database -yash
    email: "",
    password: "",
    confirmpassword: ""
  };

  OnChange = event => {
    //change of state when user enters in the text field -yash
    this.setState({ [event.target.name]: event.target.value });
  };
  componentDidMount() {
    //checking if token is not present -yash
    if (localStorage.getItem("token")) {
      this.props.history.push("/");
    }
  }
  onSubmitClick = e => {
    //to prevent values to be shown on the searchbar -yash
    e.preventDefault();
    let a = this.state.password;
    let b = this.state.confirmpassword;
    // eslint-disable-next-line
    //compare if password and confirm password are the same -yash
    if (a == b) {
      let user = {
        email: this.state.email,
        password: this.state.password
      };
      //function resetPassword present in actions/user -yash
      this.props.resetPassword(user);
    } else {
      console.log("error");
    }
  };
  render() {
    return (
      //start of div -yash
      <div>
        {/* start of form -yash */}
        <form id="resetpasswordform">
          <fieldset>
            {/* start of fieldset -yash */}

            {/* start and end of header -yash */}
            <h1 className="resetpasswordheader">Reset Password</h1>

            {/* start of input field for email -yash */}
            <input
              className="resetpassword"
              type="text"
              name="email"
              placeholder="Enter email"
              onChange={this.OnChange}
              value={this.state.email}
            />
            {/* end of input field -yash */}

            {/* start of input field for password -yash */}
            <input
              className="resetpassword"
              type="password"
              name="password"
              placeholder="Password"
              onChange={this.OnChange}
              value={this.state.password}
            />
            {/* end of input field -yash */}

            {/* start of input field for confirm password -yash */}
            <input
              className="resetpassword"
              type="password"
              name="confirmpassword"
              placeholder="Confirm Password"
              onChange={this.OnChange}
              value={this.state.confirmpassword}
            />
            {/* end of input field -yash*/}

            {/* start of button reset password -yash*/}
            <button
              onChange={this.onChange}
              onClick={this.onSubmitClick}
              className="resetpassword-button"
            >
              Update Password
            </button>
            {/* end of button -yash*/}
          </fieldset>
          {/* end of fieldset -yash */}
        </form>
        {/* end of form -yash */}
      </div>
      // end of div -yash
    );
  }
}

const mapStateToProps = state => ({
  users: state.userReducer.users
});

export default connect(
  mapStateToProps,
  //call to reset password function -yash
  { resetPassword }
)(ResetPassword);
