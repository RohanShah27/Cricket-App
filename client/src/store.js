import { createStore, applyMiddleware, compose } from "redux";

import thunk from "redux-thunk";
import rootReducer from "./reducers";

const initialState = {};
//think will give all redducer actions to the middleware object
const middleware = [thunk];
// will give access to the extension
//   //We do this to check if chrome has the redux extension or not if not then it installs the devtoolextension in chrome
const composeEnhancers =
  typeof window === "object" && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
    ? window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({
        // Specify extensionâ€™s options like name, actionsBlacklist, actionsCreators, serialize...
      })
    : compose;
const enhancer = composeEnhancers(applyMiddleware(...middleware));
const store = createStore(rootReducer, initialState, enhancer);
export default store;
