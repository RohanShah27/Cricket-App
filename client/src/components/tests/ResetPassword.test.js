import React from "react";
import { mount } from "enzyme";
import ResetPassword from "../ResetPassword";
import { Provider } from "react-redux";
import store from "../../store";

// const AddNewAdmin = jest.fn();
const wrapper = mount(
  <Provider store={store}>
    <ResetPassword resetpassword={ResetPassword} history={[]} />
  </Provider>
);

describe("Test ResetPassword Component", () => {
  it("should render the component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should have exactly three input fields", () => {
    expect(wrapper.find("input").length).toBe(3);
  });

  it("should have exactly one button", () => {
    expect(wrapper.find("button").length).toBe(1);
  });

  it("should have exactly Reset password text on Button", () => {
    expect(wrapper.find("button").text()).toEqual("Update Password");
  });

  it("should have Email placeholder on email input field", () => {
    expect(
      wrapper
        .find("input")
        .at(0)
        .props().placeholder
    ).toBe("Enter email");
  });
});

it("should have Password placeholder on password input field", () => {
  expect(
    wrapper
      .find("input")
      .at(1)
      .props().placeholder
  ).toBe("Password");
});

it("should have Confirm Password placeholder on confirmpassword input field", () => {
  expect(
    wrapper
      .find("input")
      .at(2)
      .props().placeholder
  ).toBe("Confirm Password");
});
