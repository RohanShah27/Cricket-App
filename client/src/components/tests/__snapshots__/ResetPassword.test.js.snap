// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ResetPassword Component should render the component 1`] = `ReactWrapper {}`;
